src/main/java/xades4j/providers/impl/HttpTimeStampTokenProvider.java

```diff
@@ -148,27 +55,41 @@ private InputStream getResponse(byte[] encodedRequest) throws TimeStampTokenGene
             out.write(encodedRequest);
             out.flush();
 
-            if (connection.getResponseCode() != HttpURLConnection.HTTP_OK)
-            {
+            if (connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
                 throw new TimeStampTokenGenerationException(String.format("TSA returned HTTP %d %s", connection.getResponseCode(), connection.getResponseMessage()));
             }
 
-            // TODO do we need to invoke connection.disconnect()?
-            return new BufferedInputStream(connection.getInputStream());
-        } catch (IOException ex)
-        {
+            try (BufferedInputStream input = new BufferedInputStream(connection.getInputStream())) {
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
Closing the connection will make the input stream unusable, I would expect an exception to be thrown when somebody tries to read from it. So returning the input stream would effectively make impossible to close the connection in the HttpTimeStampTokenProvider class itself.